elasticai/creator/ir/ir_data.py

```diff
@@ -0,0 +1,33 @@
+from .ir_data_meta import IrDataMeta
+from .attributes_decorator import AttributesDescriptor
+from .attribute import Attribute
+from collections.abc import Callable
+import sys
+
+_minor = sys.version_info[1]
+if _minor >= 12:  # Self was introduced in 3.12
+    from typing import Self
+else:
+    from typing import TypeVar
+
+    Self = TypeVar("Self", bound="IrData")  # type: ignore
```

Comment:
Should we really do something like this? Or should we just raise the required python version. Because if you also want to use this feature in other modules as well you always have to check...